frontend/src/metabase/visualizations/lib/apply_tooltips.js

```diff
@@ -193,56 +193,48 @@ function applyChartTooltips(
             // the normal `cols` array, which is just the cols used in the graph axes; look in `_rawCols`
             // for any other columns. If we find them, add them at the end of the `data` array.
             //
-            // To find the actual row where data is coming from is somewhat overcomplicated because i
-            // seems to follow a strange pattern that doesn't directly correspond to the rows in our
-            // data. Not sure why but it appears values of i follow this pattern:
+            // It appears values of i follow this pattern:
             //
-            // seems to follow a strange pattern that doesn't directly correspond to the rows in our
-            // data. Not sure why but it appears values of i follow this pattern:
-            //
-            //  [Series 1]  i = 7   i = 8   i = 9  i = 10   i = 11
-            //  [Series 0]  i = 1   i = 2   i = 3  i = 4    i = 5
+            //  [Series 1]  i = 5   i = 6   i = 7  i = 8   i = 9
+            //  [Series 0]  i = 0   i = 1   i = 2   i = 3   i = 4
```

Comment:
This indexing is clearly different from what had been previously observed and documented. I wondered if we either updated something or if different aggregations could weirdly trigger a different pattern here?

Reply:
This was some totally wacky code that I actually just replaced in another PR that was unfortunately not merged yet: https://github.com/metabase/metabase/pull/10057/files#diff-fdf08c963b9d1a8adf792e1c8354c389R201

You may want to take a look at that and see if it helps simplify things.

Reply:
🤦‍♂ that's much better. I'll review that one now, and I expect it will fix this issue.